plugins/outputs/application_insights/application_insights.go

```diff
@@ -0,0 +1,427 @@
+package application_insights
+
+import (
+	"fmt"
+	"log"
+	"math"
+	"time"
+	"unsafe"
+
+	"github.com/Microsoft/ApplicationInsights-Go/appinsights"
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/plugins/outputs"
+)
+
+type TelemetryTransmitter interface {
+	Track(appinsights.Telemetry)
+	Close() <-chan struct{}
+}
+
+type DiagnosticsMessageSubscriber interface {
+	Subscribe(appinsights.DiagnosticsMessageHandler) appinsights.DiagnosticsMessageListener
+}
+
+type ApplicationInsights struct {
+	InstrumentationKey      string
+	Timeout                 internal.Duration
+	EnableDiagnosticLogging bool
+	ContextTagSources       map[string]string
+	diagMsgSubscriber       DiagnosticsMessageSubscriber
+	transmitter             TelemetryTransmitter
+	diagMsgListener         appinsights.DiagnosticsMessageListener
+}
+
+var (
+	sampleConfig = `
+## Instrumentation key of the Application Insights resource.
+instrumentationKey = "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxx"
+
+## Timeout on close. If not provided, will default to 5s. 0s means no timeout (not recommended).
+# timeout = "5s"
+
+## Determines whether diagnostic logging (for Application Insights endpoint traffic) is enabled. Default is false.
+# enableDiagnosticLogging = "true"
+
+## ContextTagSources dictionary instructs the Application Insights plugin to set Application Insights context tags
+## using metric properties. In this dictionary keys are names of metric properties, values are Application Insights context tags to set using the corresponding metric property.
+## For example:
+# [outputs.application_insights.contextTagSources]
+# kubernetes_container_name = "ai.cloud.role"
+# kubernetes_pod_name = "ai.cloud.roleInstance"
+## will set the ai.cloud.role context tag to the value of kubernetes_container_name property (if present), 
+## and the ai.cloud.roleInstance context tag to the value of kubernetes_pod_name property.
+`
+	is32Bit        bool
+	is32BitChecked bool
+)
+
+func (a *ApplicationInsights) SampleConfig() string {
+	return sampleConfig
+}
+
+func (a *ApplicationInsights) Description() string {
+	return "Send telegraf metrics to Azure Application Insights"
+}
+
+func (a *ApplicationInsights) Connect() error {
+	if a.Timeout.Duration == 0 {
```

Comment:
So it's never possible that timeout is 0s? The doc said it can be set to 0s meaning no timeout.

Reply:
Agreed, this is not the right place to do apply the default. Will fix